src/librustc_mir/borrow_check/nll/liveness_map.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
```

Comment:
We should add a module doc comment here explaining what is going on. Something like this:

---

For the NLL computation, we need to compute liveness, but only for those local variables whose types contain regions. The others are not of interest to us. This file defines a new index type (`LocalWithRegion`) that indexes into a list of "variables whose type contain regions". It also defines a map from `Local` to `LocalWithRegion` and vice versa -- this map can be given to the liveness code so that it only operates over variables with regions in their types, instead of all variables.